### PR TITLE
impl basic structs for plugins

### PR DESCRIPTION
**Pull request by ameijer**
## What does this PR change?
* roughs-in the basic structs that are going to be passed around for plugin costs

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* no impact, just defining the ways that costs will be passed around 

## Does this PR address any GitHub or Zendesk issues?
* Closes ...

## How was this PR tested?
* 

## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

**Review by nik-kc — APPROVED** [review]
Looks good to me!

**Review by cliffcolvin — APPROVED** [review]

File: core/pkg/model/custom_cost_request.go
Comment (michaelmdresser) [inline core/pkg/model/custom_cost_request.go:1]: I'm going to chime in for @mbolt35 here -- please don't name files with underscores unless taking advantage of the `_test.go` or `_<platform>.go` special behavior (https://stackoverflow.com/questions/25161774/what-are-conventions-for-filenames-in-go). 

This particular file name should behave fine, but we like to encourage avoiding underscores for consistency.

File: core/pkg/model/custom_cost_request.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+	"github.com/opencost/opencost/core/pkg/opencost"
+)
+
+type CustomCostRequest struct {
+	TargetWindow *opencost.Window
+	Resolution   time.Duration
+}
+
+func (c *CustomCostRequest) GetTargetWindow() *opencost.Window {
```
Comment (michaelmdresser) [inline core/pkg/model/custom_cost_request.go:14]: Why have getters if all fields are exported and you aren't copying them?

File: core/pkg/model/custom_cost_response.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	Costsource string
+	Domain     string
+	Version    string
+	Currency   string
+	Window     opencost.Window
+	Costs      []*CustomCost
+	Errors     []error
+}
+
+type CustomCost struct {
```
Comment (michaelmdresser) [inline core/pkg/model/custom_cost_response.go:16]: Would you add godoc strings at least to the structs explaining what they are? E.g. the FOCUS inspiration etc.

File: core/pkg/model/custom_cost_request.go
Comment (ameijer) [reply core/pkg/model/custom_cost_request.go:1]: fixed, thanks for tip

File: core/pkg/model/custom_cost_request.go (hunk 1 above)
Comment (ameijer) [reply core/pkg/model/custom_cost_request.go:14]: for the RPC, I think they have to implement an interface. I have just added that interface definition

File: core/pkg/model/custom_cost_response.go (hunk 2 above)
Comment (ameijer) [reply core/pkg/model/custom_cost_response.go:16]: will do. These will come in via the next PR

File: core/pkg/model/custom_cost_request.go (hunk 1 above)
Comment (michaelmdresser) [reply core/pkg/model/custom_cost_request.go:14]: Ah, this probably answers the questions I just asked on the other PR. Perhaps we continue the discussion there; I'm curious about this requirement

**Pull request merged**